hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go

```diff
@@ -1551,6 +1551,64 @@ func TestPauseHostedControlPlane(t *testing.T) {
 	}
 }
 
+func TestHostedControlPlaneNoSecret(t *testing.T) {
+	fakeNonExistantSecret := "non-existant-secret"
+	fakeExistantSecret := "cluster-fake-pull-secret"
+	fakeHCPName := "cluster-fake"
+	fakeHCPNamespace := "master-cluster-fake"
+	testsCases := []struct {
+		name                    string
+		goodInputObjects        []crclient.Object
+		wrongInputObjects       []crclient.Object
+		inputHostedControlPlane *hyperv1.HostedControlPlane
+	}{
+		{
+			name:                    "if a hostedControlPlane exists then reference a non existant pull secret and wait to reconcile but will not progress.",
+			inputHostedControlPlane: manifests.HostedControlPlane(fakeHCPNamespace, fakeHCPName),
+			wrongInputObjects: []crclient.Object{
```

Comment:
you have a `[]struct {` but there's only on item?

It'd be probably clearer to have two test cases one with the right input and another one with the bad input, you can have an expectedError as well to support the different test cases.